pkg/sanity/sanity.go

```diff
@@ -17,26 +17,58 @@ limitations under the License.
 package sanity
 
 import (
+	"fmt"
+	"io/ioutil"
 	"sync"
 	"testing"
 
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
+	yaml "gopkg.in/yaml.v2"
 )
 
 var (
-	osdAddress   string
-	volumeDriver string
-	lock         sync.Mutex
+	osdAddress        string
+	volumeDriver      string
+	cloudBackupConfig *CloudBackupConfig
+	lock              sync.Mutex
 )
 
+// CloudBackupConfig struct for cloud backup configuration
+type CloudBackupConfig struct {
+	//CloudProvider string `yaml:"providers"`
+	// map[string]string is volume.VolumeParams equivalent
+	CloudProviders map[string]map[string]string
+}
+
 // Test will test the CSI driver at the specified address
-func Test(t *testing.T, address, driver string) {
+func Test(t *testing.T, address, driver, cloudBackupConfigPath string) {
 	lock.Lock()
 	defer lock.Unlock()
 
+	cfg, err := CloudProviderConfigParse(cloudBackupConfigPath)
```

Comment:
You probably can check if `len(cloudBackupConfigPath) == 0` and not make this call at all, right?

Reply:
makes sense , parse the file only if its provided.